pykwasm/src/pykwasm/run_wasm.py

```diff
@@ -0,0 +1,557 @@
+    patched_config_kore = PatternWriter(config_kore)
+
+    # log input kore
+    if DEBUG:
+        with open(wasm_file.name + '.input.kore','w') as f:
+            patched_config_kore.write(f)
+
+    # run the config
+    proc_data = runner.run_process(patched_config_kore, term=True, expand_macros=False)
+
+    # print the result
+    print(proc_data.stdout)
+    if proc_data.returncode != 0 or DEBUG:
+        print(proc_data.stderr, file=sys.stderr)
+    proc_data.check_returncode()
+
+class DepthChange(Enum):
+    UP = 1
+    DOWN = -1
+    PRINT = 0
+
+def pattern_write(pat: Pattern, output: IO[str], pretty=True) -> None:
+    """Serialize pattern to kore; used for monkey patch on Pattern object because default write function will blow the stack"""
+
+    if pretty:
+        UP, DOWN, PRINT = DepthChange.UP, DepthChange.DOWN, DepthChange.PRINT
+    else:
+        UP, DOWN, PRINT = ['']*3
+    not_first_term = False
+    print_spacer = False
+    depth = 0
+    stack = [pat]
+
+    # TODO: fix bug with workitems order
+
+    def push(*items):
+        for item in reversed(items):
+            if isinstance(item, tuple):
+                if len(item) > 1:
+                    for subitem in reversed(item[1:]):
+                        stack.append(subitem)
+                        stack.append(',')
+                if len(item) > 0:
+                    stack.append(item[0])
+            elif isinstance(item, (str, DepthChange)):
+                stack.append(item)
+            else:
+                raise ValueError(f"Unexpected item type: {type(item)}")
+
+    while len(stack) > 0:
+        pat = stack.pop()
+        if isinstance(pat, str):
+            if print_spacer:
+                if not_first_term: output.write('\n' + depth*' ')
+                not_first_term = True
+                print_spacer = False
+            output.write(pat)
+        elif isinstance(pat, syntax.App):
+            push(PRINT, pat.symbol, '{', pat.sorts, '}(', UP, pat.args, DOWN, ')')
+        elif isinstance(pat, syntax.Assoc):
+            push(PRINT, pat.kore_symbol(), '{}(', UP, self.app, DOWN, ')')
+        elif isinstance(pat, syntax.MLPattern):
+            push(PRINT, pat.symbol(), '{',  pat.sorts, '}(', pat.ctor_patterns, ')')
+        elif isinstance(pat, syntax.SortApp):
+            push(pat.name, '{', pat.sorts, '}')
+        elif isinstance(pat, DepthChange):
+            depth += pat.value
+            if pat == PRINT:
+                print_spacer = True
+        else:
+            pat.write(output)
+
+class PatternWriter:
+    def __init__(self, pat: Pattern, pretty=False):
+        self.pat = pat
+        self.pretty = pretty
+
+    def write(self, output: IO[str]):
+        if isinstance(self.pat, (syntax.App, syntax.SortApp, syntax.Assoc, syntax.MLPattern)):
+            pattern_write(self.pat, output, self.pretty)
+        else:
+            self.pat.write(output)
+
+def debug(pat) -> str:
+        if isinstance(pat, str):
```

Comment:
It's probably ok to leave as-is, but this indentation level does not match the other functions.

Reply:
Fixed! Thanks!